lib/annotations2triannon/open_annotation.rb

```diff
@@ -18,16 +18,19 @@ class OpenAnnotation
     # @param id [UUID|URI|String] to identify an open annotation
     def initialize(graph=RDF::Graph.new, id=nil)
       @@agent ||= Annotations2triannon::AGENT
-      raise ArgumentError, 'graph must be RDF::Graph instance' unless graph.instance_of? RDF::Graph
-      @graph = graph
-      @id = id.nil? ? get_id : RDF::URI.parse(id)
-      insert_annotation unless is_annotation?
+      set_graph(graph)
+      set_id(id) # must be set after setting the graph
+      insert_annotation unless is_annotation? # useful when graph.empty?
+    end
+
+    # @see #set_id
+    def id=(id)
```

Comment:
@darrenleeweber you could also write it as:

```ruby
def initialize id
  ...
  self.id = id
end
```

Maybe another way to refactor this would be, e.g.

```ruby
def initialize id
  @id = parse_id(id)
end

def id=(id)
  @id = parse_id(id)
end
```

That way, the parsing logic is clearly separated from the setting-the-instance-variable, possibly making both easier to test.